### PR TITLE
Update EIA hydro generation data and remove vresutils dependency

### LINKED ISSUE #324: Workflow improvement: Support new EIA data format in build_hydro
The EIA data format from our datasource changed. Suggestions
- Automatic download of EIA data OR update the EIA .csv file with hydro generation to new dataformat
- substitute the `vresutils` dependency in `build_hydro` by writing the `pd.read_csv` statement directl.y

_Originally posted by @euronion in https://github.com/PyPSA/pypsa-eur/pull/321#issuecomment-1059728440_

### PR DESCRIPTION
**Pull request by fneum**
Closes #324.

## Changes proposed in this Pull Request

- [ ] help wanted to automate download without API

## Checklist

- [ ] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [ ] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes.

### CODE REVIEW

File: scripts/build_hydro_profile.py [hunk 1]
```diff
@@ -64,7 +64,29 @@
 
 import atlite
 import geopandas as gpd
-from vresutils import hydro as vhydro
+import pandas as pd
+
+import country_converter as coco
+cc = coco.CountryConverter()
```
Comment (euronion) [inline scripts/build_hydro_profile.py:70]: Change of dependency in `environment.yaml` from PyPSA-EUR-Sec dep to PyPSA-EUR dep.

File: scripts/build_hydro_profile.py [hunk 2]
```diff
@@ -64,7 +64,29 @@
...
+
+    # in billion kWh/a = TWh/a
+    df = pd.read_csv(fn, skiprows=2, index_col=1, na_values=[u' ','--']).iloc[1:, 1:]
+    df.index = df.index.str.strip()
+
+    df.loc["Germany"] = df.filter(like='Germany', axis=0).sum()
+    df.loc["Serbia"] += df.loc["Kosovo"]
+    df = df.loc[~df.index.str.contains('Former')]
+
+    df.index = cc.convert(df.index, to='iso2')
```
Comment (euronion) [inline scripts/build_hydro_profile.py:83]: This throws three warnings
```
Europe not found in regex
Germany, East not found in regex
Germany, West not found in regex
```

Maybe also exclude those similarly to line 81?

**Review by euronion — CHANGES_REQUESTED** [review]
Any idea why the historic EIA values changed? It's mostly a few percent (e.g. Austria), but some of the value changes are > 15% (mostly for countries with very low / low hydro generation)

image

Also needs an update to the RELEASE_NOTES (less databundle dependency, more data in repo).

Else LGTM.

**Comment by fneum** [comment]
No idea where the difference in EIA data come from.

**Pull request merged**